NYU Press.js

```diff
@@ -0,0 +1,276 @@
+{
+	"translatorID": "2ca2c054-7084-458f-b493-86e388284720",
+	"label": "NYU Press",
+	"creator": "Nathan Kim",
+	"target": "https?://nyupress\\.org/([0-9]){13}",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-05-12 00:33:10"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2022 Nathan Kim
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb() {
+	return "book";
+}
+
+function doWeb(doc, url) {
+	var item = new Zotero.Item("book");
+	item.date = doc.querySelector(".sp__published").textContent.replace("Published: ", "");
+	item.ISBN = doc.querySelector(".sp__isbn13").textContent;
+
+	for (let roleNode of doc.querySelectorAll('.sp__the-author')) {
+		let role;
+		if (roleNode.textContent.match("Edited")) {
+			role = "editor";
+		}
+		else if (roleNode.textContent.match("Foreword|Introduction")) {
+			role = "contributor";
+		}
+		else {
+			role = "author";
+		}
+		for (let authorNode of roleNode.querySelectorAll(".sp__author-link")) {
+			item.creators.push(ZU.cleanAuthor(authorNode.textContent, role));
+		}
+	}
+
+	item.numPages = doc.querySelector(".sp__the-pages").textContent.replace(" Pages", "");
+	item.abstractNote = doc.querySelector(".tabs__panel--description p").textContent;
+	let series = doc.querySelector(".book-wrapper__top-section--details h3");
+	item.series = series ? series.textContent : '';
+	item.publisher = doc.querySelector(".sp__the-publisher").textContent.replace("Published by: ", "");
+
+	item.url = url;
```

Comment:
remove -- the URL field should only be used for full-text sources (which I assume these almost never are) -- it'll be used in citations. You'll save the link via the Snapshot.